code/game/gamemodes/changeling/powers/transform.dm

```diff
@@ -11,36 +11,93 @@
 	name = "flesh"
 	flags = NODROP
 
+/obj/item/clothing/glasses/changeling/attack_hand(mob/user)
+	if(loc == user && user.mind && user.mind.changeling)

```

Comment:
No it wouldn't, because the flag removal it still behind the user midn changeling check.

I'm saying remove the flag in the attack_hand and set DROPDEL (you can probably just set DROPDEL at compile time for them), but the NODROP will only be unset when this attack hand check passes